SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -485,11 +490,33 @@ bool FLocalDeploymentManager::GetServiceStatus()
 		bLocalDeploymentRunning = false;
 		return false;
 	}
-	else
+	else if (ServiceStatusResult.Contains(TEXT("Local API service is running")))
 	{
-		UE_LOG(LogSpatialDeploymentManager, Verbose, TEXT("Spatial service running."));
-		bSpatialServiceRunning = true;
-		return true;
+		// Get the project file path and ensure it matches the one for the currently running project.
+		FString SpatialServicePath;
+		if (ServiceStatusResult.Split(TEXT("project file path: "), nullptr, &SpatialServicePath))
+		{
+			// Remove the trailing '" cli-version' that comes with non-interactive 'spatial' calls.
+			SpatialServicePath.Split(TEXT("\" cli-version"), &SpatialServicePath, nullptr);
```

Comment:
Might be a little cleaner if the code that initialises directory names is in a separate function, that returns bool as a result of (CurrentProjectSpatialPath == SpatialServicePath) comparison.